polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala

```diff
@@ -42,41 +41,19 @@ class PolyKernel private[kernel] (
   private val launchingKernel = Semaphore[IO](1).unsafeRunSync()
   private val kernels = new ConcurrentHashMap[String, LanguageKernel[IO]]()
 
-  final protected def dependencyClassPath: Seq[URL] = dependencies.toSeq.flatMap(_._2).collect {
-    case (_, file) if (file.getName endsWith ".jar") && file.exists() => file.toURI.toURL
-  }
-
-  /**
-    * @return The complete classpath, including dependencies and extraClassPath
-    */
-  final protected def classPath: Seq[URL] = dependencyClassPath ++ extraClassPath
-
-  /**
-    * The class loader which loads the dependencies
-    */
-  protected lazy val dependencyClassLoader: URLClassLoader =
-    new LimitedSharingClassLoader(
-      "^(scala|javax?|jdk|sun|com.sun|com.oracle|polynote|org.w3c|org.xml|org.omg|org.ietf|org.jcp|org.apache.spark|org.apache.hadoop|org.codehaus)\\.",
-      dependencyClassPath,
-      parentClassLoader)
-
-  /**
-    * The class loader which loads the JVM-based notebook cell classes
-    */
-  protected lazy val notebookClassLoader: AbstractFileClassLoader = new AbstractFileClassLoader(outputDir, dependencyClassLoader)
 
   /**
     * The symbol table, which stores all the currently existing runtime values
     */
-  protected lazy val symbolTable = new RuntimeSymbolTable(global, notebookClassLoader, statusUpdates)
+  protected lazy val symbolTable = new RuntimeSymbolTable(globalInfo, statusUpdates)
 
   /**
     * The task queue, which tracks currently running and queued kernel tasks (i.e. cells to run)
     */
   protected lazy val taskQueue: TaskQueue = TaskQueue(statusUpdates).unsafeRunSync()
 
   private def runPredef(kernel: LanguageKernel[IO], language: String): IO[Unit] =
-    kernel.predefCode.fold(IO.unit) {
+    kernel.predefCode.fold(kernel.init()) {
```

Comment:
At some point we should probably have this just unconditionally run `kernel.init()`, and then say `kernel.init()` has to run its own predef (and remove `predefCode`). This is good for now though.

Reply:
Ah blerg, I forgot that Option.fold isn't like a normal fold in that it doesn't execute that zero case if predefCode is a Some... 🤦‍♂️ 

Agree that it should be ok for now though